src/components/common/Button.js

```diff
@@ -0,0 +1,108 @@
+import React from "react";
+import {
+	StyleSheet,
+	Text,
+	Pressable,
+	View,
+	ActivityIndicator
+} from "react-native";
+
+import PropTypes from "prop-types";
+
+import Icon from "react-native-vector-icons/MaterialIcons";
+import { COLOR_PALETTE } from "../../utils/constants";
+
+export const BUTTON_TYPES = {
+	PRIMARY: "primary",
+	TEXT: "text"
+};
+const Button = ({
+	text,
+	onPress,
+	style = {},
+	textStyle = {},
+	buttonTestID,
+	icon,
+	type = BUTTON_TYPES.PRIMARY,
+	isLoading
+}) => {
+	const getStyleForButtonType = () =>
+		type === BUTTON_TYPES.PRIMARY ? styles.primaryButton : {};
+
+	const renderButton = () =>
+		isLoading ? (
+			<ActivityIndicator
+				style={styles.loader}
+				size="large"
+				color={COLOR_PALETTE.lightblue}
+			/>
+		) : (
+			<Pressable
+				style={{ ...styles.button, ...getStyleForButtonType(), ...style }}
+				onPress={onPress}
+				testID={buttonTestID}
+				disabled={isLoading}
+			>
+				{icon && (
```

Comment:
the activity indicator should go inside the pressable. So it should be something like
```
<pressable>
{isLoading ? renderActivityIndicator() : renderButton()}
</Pressable>
```